src/main/java/com/fvv/bookstore/view/BookViewImpl.java

```diff
@@ -0,0 +1,184 @@
+			book.setAuthor(JOptionPane.showInputDialog("Insert the author"));
+			book.setPrice(
+					Double.parseDouble(JOptionPane.showInputDialog("Insert the price")));
+			book.setIsbn(
+					Integer.parseInt(JOptionPane.showInputDialog("Insert the isbn")));
+			book.setPublisher(JOptionPane.showInputDialog("Insert the publisher"));
+			book.setGenre(JOptionPane.showInputDialog("Insert the genre"));
+			
+			this.bookController.validateBook(book);
+			boolean isSuccessful = this.bookController.addBook(book);
+			
+			if (isSuccessful) {
+				JOptionPane.showMessageDialog(null, "Book added successfuly!");
+			} else {
+				JOptionPane.showMessageDialog(null, "Error to add the book!");
+			}			
+		} catch (NumberFormatException e) {
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, "Problems to convert the value:\n" 
+					+ e.getMessage(), "Error", JOptionPane.ERROR_MESSAGE);
+		} catch (ControllerException e) {
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, e.getMessage(), "Error", 
+					JOptionPane.ERROR_MESSAGE);
+		} catch (BookValidationException e) {
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, e.getMessage(), "Error", 
+					JOptionPane.ERROR_MESSAGE);
+		}		
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void listBooks() {
+		try {
+			StringBuilder sb = new StringBuilder();
+			List<Book> books = this.bookController.listBooks();
+			if(books != null && !books.isEmpty()) {
+				for(Book b : books) {
+					sb.append(b).append("\n");
+				}
+				JOptionPane.showMessageDialog(null, sb.toString(), "Listing All Books", 
+						JOptionPane.PLAIN_MESSAGE);
+			} else {
+				JOptionPane.showMessageDialog(null, "There is no itens to show!");
+			}			
+		} catch (ControllerException e) {
+			e.printStackTrace();
+			JOptionPane.showMessageDialog(null, e.getMessage(), "Error", 
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void updateBook() {
+		try {
+			Book book = new Book();
+			Long idUp = Long.parseLong(JOptionPane.showInputDialog("Insert the ID to update"));
+			Book findBook = this.findBook(idUp);
+			
+			if(findBook != null) {
+				book.setId(idUp);
+				book.setTitle(JOptionPane.showInputDialog("Insert the title"));
+				book.setPublicationYear(
+						Integer.parseInt(JOptionPane.showInputDialog("Insert the publication year")));
+				book.setEditionNumber(
+						Integer.parseInt(JOptionPane.showInputDialog("Insert the edition number")));
+				book.setAuthor(JOptionPane.showInputDialog("Insert the author"));
+				book.setPrice(
+						Double.parseDouble(JOptionPane.showInputDialog("Insert the price")));
+				book.setIsbn(
+						Integer.parseInt(JOptionPane.showInputDialog("Insert the isbn")));
+				book.setPublisher(JOptionPane.showInputDialog("Insert the publisher"));
+				book.setGenre(JOptionPane.showInputDialog("Insert the genre"));
+				
+				this.bookController.validateBook(book);
```

Comment:
as stated earlier, this should not be here, you need to call this validation in the controller class